README.md

```diff
@@ -16,3 +16,203 @@ This repository contains an implementation of a SignalR server and a SignalR cli
+
+```go
+package main
+
+import "github.com/philippseith/signalr"
+
+type AppHub struct {
+    signalr.Hub
+}
+```
+
+Add functions with your custom hub type as a receiver.
+
+```go
+func (h *AppHub) SendChatMessage(message string) {
+    h.Clients().All().Send("chatMessageReceived", message)
+}
+```
+
+These functions must be public so that they can be seen by the signalr server package but can be invoked client-side as lowercase message names.  We'll explain setting up the client side in a moment, but as a preview, here's an example of calling our `AppHub.SendChatMessage(...)` method from the client:
+
+```js
+    // javascript snippet invoking that AppHub.Send method from the client
+    connection.invoke('sendChatMessage', val);
+```
+
+The `signalr.HubInterface` contains a pair of methods you can implement to handle connection and disconnection events.  `signalr.Hub` contains empty implementations of them to satisfy the interface, but you can "override" those defaults by implementing your own functions with your custom hub type as a receiver:
+
+```go
+func (c *chat) OnConnected(connectionID string) {
+    fmt.Printf("%s connected\n", connectionID)
+}
+
+func (c *chat) OnDisconnected(connectionID string) {
+   fmt.Printf("%s disconnected\n", connectionID)
+}
+```
+
+#### Serve with http.ServeMux
+
+```go
+import "net/http"
+
+func runHTTPServer() {
+    address := 'localhost:8080'
+    
+    // create an instance of your hub
+    hub := AppHub{}
+	
+    // build a signalr.Server using your hub
+    server, _ := signalr.NewServer(context.TODO(), signalr.SimpleHubFactory(hub))
+    
+    // configure global signalr options
+    signalr.KeepAliveInterval(2*time.Second),
+    signalr.Logger(kitlog.NewLogfmtLogger(os.Stderr), true))
+    
+    // create a new http.ServerMux to handle your app's http requests
+    router := http.NewServeMux()
+    
+    // ask the signalr server to map it's server
+    // api routes to your custom baseurl
+    server.MapHTTP(router, "/chat")
+
+    // in addition to mapping the signalr routes
+    // your mux will need to serve the static files
+    // which make up your client-side app, including
+    // the signalr javascript files. here is an example
+    // of doing that using a local `public` package
+    // which was created with the go:embed directive
+    // 
+    // fmt.Printf("Serving static content from the embedded filesystem\n")
+    // router.Handle("/", http.FileServer(http.FS(public.FS)))
+    
+    // bind your mux to a given address and start handling requests
+    fmt.Printf("Listening for websocket connections on http://%s\n", address)
+    if err := http.ListenAndServe(address, router); err != nil {
+        log.Fatal("ListenAndServe:", err)
+    }
+}
+```
+
+### Client side
+
+#### Grab copies of the signalr scripts
```

Comment:
Many web devs are using package managers, so something like `npm install @microsoft/signalr` should be mentioned.

Reply:
good idea, I'll fix that.

Reply:
replaced that section with a table showing the commands for npm, yarn, and libman